lms/djangoapps/grades/models.py

```diff
@@ -47,11 +58,12 @@ def to_json(self):
         if self._json is None:
             sorted_blocks = sorted(self, key=attrgetter('locator'))
             list_of_block_dicts = [block._asdict() for block in sorted_blocks]
-            course_key = next(iter(self)).locator.course_key  # all blocks are from the same course
+            course_key_string = self._get_course_key_string() # all blocks are from the same course
+
             for block_dict in list_of_block_dicts:
                 block_dict['locator'] = unicode(block_dict['locator'])  # BlockUsageLocator is not json-serializable
             data = {
-                'course_key': unicode(course_key),
+                'course_key': course_key_string,
```

Comment:
It doesn't have to, because there are no blocks.

On Fri, Sep 9, 2016 at 3:53 PM, Nimisha Asthagiri <notifications@github.com>
wrote:

> In lms/djangoapps/grades/models.py
> <https://github.com/edx/edx-platform/pull/13437#discussion_r78238924>:
>
> >              for block_dict in list_of_block_dicts:
> >                  block_dict['locator'] = unicode(block_dict['locator'])  # BlockUsageLocator is not json-serializable
> >              data = {
> > -                'course_key': unicode(course_key),
> > +                'course_key': course_key_string,
>
> You probably need to update from_json to check for the Null case. I don't
> think CourseKey.from_string works with None.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/edx/edx-platform/pull/13437/files/9a09baed2d94cfe50c211f8be18727083c13cd42..40a4b67eaf52072de8efbcaa506514dc97a064af#r78238924>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAEuriGaD5YkgbB5rrC3gaUngGiI8k_hks5qoblLgaJpZM4J5UGh>
> .
>